test/test_executable.py

```diff
@@ -0,0 +1,21 @@
+import os
+import tempfile
+
+import clang_tidy
+
+
+def test_executable_file():
+    exe = clang_tidy._get_executable("clang-tidy")
+    assert os.path.exists(exe), "'clang-tidy' executable is missing"
+    assert os.access(exe, os.X_OK), "'clang-tidy'program is not executable"
+
+
+def test_include_iostream():
+    fd, compilation_unit = tempfile.mkstemp(suffix=".cpp")
+    os.close(fd)
+    with open(compilation_unit, "w") as ostr:
+        ostr.write("#include <iostream>\n")
+    try:
+        assert clang_tidy._run("clang-tidy", compilation_unit) == 0
+    finally:
+        os.remove(compilation_unit)
```

Comment:
Using a pytest fixture to provide a `pathlib.Path` object for a temporary directory, 
means you don't have to worry about cleanup.

```suggestion
def test_include_iostream(tmp_path):
    compilation_unit = tmp_path / "test_iostream.cpp"
    compilation_unit.write_text("#include <iostream>\n")
    assert clang_tidy._run("clang-tidy", compilation_unit) == 0
```

Reply:
IMHO, I would ditch Python 2.
For the oldest Python 3 version to support in open-source context, I usually rely on the version of Python delivered in the oldest Ubuntu LTS. It is currently [20.04 LTS (Focal Fossa)](https://ubuntu.com/about/release-cycle), which provides Python [3.8.2](https://packages.ubuntu.com/focal/python3)

Reply:
Either way, we'd be newer than 3.4 and fine for `pathlib` :+1: